expertise/service/routes.py

```diff
@@ -0,0 +1,458 @@
+        return flask.jsonify(result), 500
+        
+    else:
+        flask.current_app.logger.debug('POST returns ' + str(result))
+        return flask.jsonify(result), 200
+
+@BLUEPRINT.route('/results', methods=['GET'])
+def results():
+    """
+    Get the results of a single submitted job with the associated job_id
+    If provided with a delete_on_get field, delete the job from the server after retrieving results
+
+    :param token: Authorization from a logged in user, which defines the set of accessible data
+    :type token: str
+
+    :param job_id: The ID of a submitted job
+    :type job_id: str
+    
+    :param delete_on_get: Decide whether to keep the data on the server after getting the results
+    :type delete_on_get: bool
+    """
+    result = {}
+
+    token = flask.request.headers.get('Authorization')
+    in_test_mode = 'IN_TEST' in flask.current_app.config.keys()
+    
+    if not token and not in_test_mode:
+        flask.current_app.logger.error('No Authorization token in headers')
+        result['error'] = 'No Authorization token in headers'
+        return flask.jsonify(result), 400
+    
+    try:
+        job_id = flask.request.args['job_id']
+        delete_on_get = flask.request.args.get('delete_on_get', 'False')
+
+        if delete_on_get.lower() == 'true':
+            delete_on_get = True
+        else:
+            delete_on_get = False
+    
+        if not in_test_mode:
+            openreview_client = openreview.Client(
+                token=token,
+                baseurl=flask.current_app.config['OPENREVIEW_BASEURL']
+            )
+        else:
+            openreview_client = mock_client()
+
+        _, profile_dir = get_profile_and_id(openreview_client)
+        if not os.path.isdir(profile_dir):
+            raise OpenReviewException('No jobs submitted since last server reboot')
+
+        # Search for scores files (only non-sparse scores)
+        search_dir = os.path.join(profile_dir, job_id)
+        file_dir, metadata_dir = get_score_and_metadata_dir(search_dir)
+
+        # Assemble scores
+        if file_dir is None:
+            raise OpenReviewException('Either job is still processing, has crashed, or does not exist')
+        else:
+            ret_list = []
+            with open(file_dir, 'r') as csv_file:
+                data_reader = reader(csv_file)
+                for row in data_reader:
+                    ret_list.append({
+                        'submission': row[0],
+                        'user': row[1],
+                        'score': float(row[2])
+                    })
+            result['results'] = ret_list
+
+            # Gather metadata
+            with open(metadata_dir, 'r') as metadata:
+                result['metadata'] = json.load(metadata)
+
+        # Clear directory
+        if delete_on_get:
+            flask.current_app.logger.error(f'Deleting {search_dir}')
+            shutil.rmtree(search_dir)
```

Comment:
Has it been discussed for how long we will store results if `delete_on_get` is not passed?

Reply:
I don't think we've discussed this - maybe one solution is to keep them for a week?

Implementation-wise I was thinking I would store the creation date of the job and have a separate thread query all existing job directories for their creation date and if the current date is one week past, clean the directory. Then the thread would sleep for 24 hours so that this check would happen daily.

Is this a reasonable solution?